pycargoebuild/cargo.py

```diff
@@ -87,14 +104,20 @@ def get_package_metadata(f: typing.BinaryIO) -> PackageMetadata:
             "pycargoebuild on one of its members instead: "
             f"{' '.join(cargo_toml['workspace']['members'])}")
 
+    workspace_pkg_meta = cargo_toml.get("workspace", {}).get("package", {})
     pkg_meta = cargo_toml["package"]
-    pkg_license = pkg_meta.get("license")
+    pkg_license = get_pkg_meta("license", pkg_meta, workspace_pkg_meta)
```

Comment:
This really looks like you want to use `functools.partial()` instead of repeating `pkg_meta, workspace_pkg_meta` every time.

Reply:
or make a class